### ISSUE TITLE
Unexpected navigation results

### ISSUE THREAD

**Issue by dbay920** [dbay920/phn-app#31]
Getting unexpected results with more/back button as well as navigation patterns on both iOS and Android

Examples:
Home -> Services -> Behavioral Health -> Autumn St. 
- tapping back button takes you to Locations view (?) instead of back through view history
- tapping back button at this point removes back button and nothing happens 

Home -> Locations -> Ashtabula Cty -> Andover Primary Care
- back button works OK until you get to Locations view, then it goes away (can't go back to Home view)
- if I tap on Home and then from Home view click on Locations it takes me to Andover Primary Care location details page instead of Locations default view. If I tap on Locations tab, same thing, Andover Primary Care Locations Detail page. I can't navigate to Locations default view. This pattern happens with Services and Providers as well.
In other words, if I go to a Location detail view, tap on Home, then tap on Locations, I am taken to the Location detail view instead of Location default view.

On iOS, if I use the More button to get to Services view then tap back button it takes me to More view but I can't go back any further since back button is removed on this view.
